### DISCUSSION TITLE
Whether to support rolling back data to a specified time point

### DISCUSSION THREAD

**Discussion by Aneuuu** [apache/kvrocks#2442] [Q&A] [answered]
Whether to support rolling back data to a specified time point

**Comment by torwig** [comment]
@Aneuuu Thank you for starting a discussion. Could you please elaborate on the topic? As far as I know, Redis supports making snapshots and restoring the state from the snapshots. Do you mean something else?

**Reply by Aneuuu** [reply]
i thisk snapshot just a time point, but between of two time point has data diff

**Comment by PragmaTwice** [comment]
You can use BGSAVE to backup all your data.

Also you can set the BGSAVE scheduler `bgsave-cron` in config file to setup regular backup.

https://kvrocks.apache.org/docs/backup/

**Reply by Aneuuu** [reply]
I want to make up for the data differences caused by the cron

**Reply by PragmaTwice** [reply, marked as answer]
No database allows you to (globally) backtrack to any time point without any cost.

It will come with huge storage expenses.

**Comment by mapleFU** [comment]
Seems this require something like PITR

Anyway this requires capture the "binlog" and keeping the logs between version 🤔 If you want you can trying to duplicate the logs for cluster and add this feature

Beside, the bgsave is also important here since keeping the cluster log is very expansive, so some syntax need to be well-design for this scenerio.
